scapy/utils.py

```diff
@@ -489,16 +489,19 @@ def do_graph(graph,prog=None,format=None,target=None,type=None,string=None,optio
     if target is None:
         if WINDOWS:
             tempfile = get_temp_file() + "." + format
-            target = "> %s" % tempfile
+            target = "-o%s" % tempfile
             start_viewer = True
         else:
             target = "| %s" % conf.prog.display
     if format is not None:
-        format = "-T %s" % format
-    w,r = os.popen2("%s %s %s %s" % (prog,options or "", format or "", target))
-    w.write(graph)
+        format = "-T%s" % format
+    proc_args = [prog, options, format, target]
+    proc_args = [x for x in proc_args if x]
+    proc = subprocess.Popen(proc_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
```

Comment:
This would change an API that has existed in Scapy for a really long time. You may want to use `subporcess.Popen("%s %s %s %s" % (prog,options or "", format or "", target), shell=True, [...])`.

Also, why you need to have `stdout=subprocess.PIPE`?

Reply:
Updated